apps/crn-server/src/data-providers/external-authors.data-provider.ts

```diff
@@ -19,4 +43,48 @@ export class ExternalAuthorSquidexDataProvider
     });
     return id;
   }
+
+  async fetch(options: FetchOptions): Promise<ListExternalAuthorResponse> {
+    const { take = 8, skip = 0 } = options;
+
+    const { queryExternalAuthorsContentsWithTotal } =
+      await this.squidexGraphqlClient.request<
+        FetchExternalAuthorsQuery,
+        FetchExternalAuthorsQueryVariables
+      >(FETCH_EXTERNAL_AUTHORS, { top: take, skip });
+
+    if (queryExternalAuthorsContentsWithTotal === null) {
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    const { total, items } = queryExternalAuthorsContentsWithTotal;
+
+    if (items === null) {
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    return {
+      total,
+      items: items.map(parseGraphQLExternalAuthor),
+    };
+  }
+
+  async fetchById(id: string): Promise<ExternalAuthorResponse> {
+    const { findExternalAuthorsContent } =
+      await this.squidexGraphqlClient.request<
+        FetchExternalAuthorQuery,
+        FetchExternalAuthorQueryVariables
+      >(FETCH_EXTERNAL_AUTHOR, { id });
+    if (!findExternalAuthorsContent) {
+      throw Boom.notFound();
```

Comment:
I think we are trying to move away from calling Boom anywhere outside of routes and controllers - this is because Boom is quite heavily REST focused. There should be a generic NotFound error you should be able to use here instead

Reply:
I think it actually looks like there's a more prevailing pattern of returning null from `fetchById` in the case of the entity not found. Will converge towards that.